programs/timing.py

```diff
@@ -70,3 +70,18 @@ def _monotonic_time_nanos_cygwin():
         k32.QueryPerformanceCounter(ctypes.byref(perf_counter))
         return perf_counter.value * NSEC_PER_SEC / perf_frequency.value
     monotonic_time_nanos = _monotonic_time_nanos_cygwin
+elif platform.system() == 'FreeBSD':
+    CLOCK_MONOTONIC = 4
+    # On FreeBSD9 and FreeBSD10
+    libc = ctypes.CDLL('libc.so.7', use_errno=True)
```

Comment:
Since this is the only difference between this case and the linux case, can you extract the common parts?